codegen/test/chart/templates/rbac.yaml

```diff
@@ -6,43 +6,98 @@
 {{- $painter := $.Values.painter }}
 {{ if $painter.enabled }}
 
+
 ---
 
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: painter-{{ default .Release.Namespace $.Values.painter.namespace }}
+  name: painter-{{ default .Release.Namespace $painter.namespace }}
   labels:
     app: painter
 rules:
 - verbs:
   - GET
+
+---
+
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: painter-{{ default .Release.Namespace $painter.namespace }}
+  labels:
+    app: painter
+subjects:
+- kind: ServiceAccount
+  name: painter
+  namespace: {{ default .Release.Namespace $painter.namespace }}
+roleRef:
+  kind: ClusterRole
+  name: painter-{{ default .Release.Namespace $painter.namespace }}
+  apiGroup: rbac.authorization.k8s.io
+
+{{- $painterNsToResources := dict }}
+{{- $painterNamespacedResources := dict }}
+{{- range $entry := $painter.namespacedRbac }}
+  {{- range $ns := $entry.namespaces }}
+    {{- set $painterNsToResources $ns (concat $entry.resources (get $painterNsToResources $ns | default list) | mustUniq) }}
+  {{- end }}
+  {{- range $resource := $entry.resources }}
+    {{- $_ := set $painterNamespacedResources $resource true }}
+  {{- end }}
+{{- end }}
+
+{{- if not (eq (len $painterNamespacedResources) 1) }}
+---
+
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: painter
```

Comment:
Where are there 2 cluster roles on this one where one has a <name>-namespace and one is just <name>

Reply:
The naming is so they don't conflict. 

We need both since the initial one is for resources that we don't support namespacing for (even if they are a namespace-scoped resource from a k8s perspective). We create the second with the eligible namespaced resources that aren't enabled from the user to be namespace-scoped.

Reply:
We need to be careful about not having the release namespace in the name, ideally we would like certain components to be installed multiple times in different namespaces. If not today that should work eventually, by not having the release namespace in the cluster role the second install will always fail because the resource exists

Reply:
We could add the release name to the second clusterrole name, and have the first clusterrole name include release name and namespace. That way it is still configured. Thiughts?